lib/closex/mock_client/mock_client.ex

```diff
@@ -0,0 +1,251 @@
+    {:error, :mock_not_found}
+  end
+
+  @doc """
+  Gets a lead custom field from CloseIO.
+
+  Returns `{:ok, lead_custom_field}`.
+
+  ## Examples
+
+    > Closex.MockClient.get_lead_custom_field(Closex.MockClient.get_lead_custom_field_id())
+    ...contents of test/fixtures/lead_custom_field.json...
+
+    iex> Closex.MockClient.get_lead_custom_field(Closex.MockClient.get_not_found_id())
+    {:error, :mock_not_found}
+  """
+  def get_lead_custom_field(id, opts \\ [])
+  def get_lead_custom_field(id = @lead_custom_field_id, opts) do
+    lead_custom_field = load("lead_custom_field.json")
+    send self(), {:closex_mock_client, :get_lead_custom_field, [id, opts]}
+    {:ok, lead_custom_field}
+  end
+  def get_lead_custom_field(id = @not_found_id, opts) do
+    send self(), {:closex_mock_client, :get_lead_custom_field, [id, opts]}
+    {:error, :mock_not_found}
+  end
+
+  @doc """
+  Gets an organization from CloseIO.
+
+  Returns `{:ok, organization}`.
+
+  ## Examples
+
+    > Closex.MockClient.get_organization(Closex.MockClient.get_organization_id())
+    ...contents of test/fixtures/organization.json...
+
+    iex> Closex.MockClient.get_organization(Closex.MockClient.get_not_found_id())
+    {:error, :mock_not_found}
+  """
+  def get_organization(id, opts \\ [])
+  def get_organization(id = @organization_id, opts) do
+    organization = load("organization.json")
+    send self(), {:closex_mock_client, :get_organization, [id, opts]}
+    {:ok, organization}
+  end
+  def get_organization(id = @not_found_id, opts) do
+    send self(), {:closex_mock_client, :get_organization, [id, opts]}
+    {:error, :mock_not_found}
+  end
+
+  @doc """
+  Gets the potential statuses of a lead from CloseIO.
+
+  Returns `{:ok, lead_statuses}`.
+
+  ## Examples
+
+    > Closex.MockClient.get_lead_statuses()
+    ...contents of test/fixtures/lead_statuses.json...
+  """
+  def get_lead_statuses(opts \\ []) do
+    lead_statuses = load("lead_statuses.json")
+    send self(), {:closex_mock_client, :get_lead_statuses, [opts]}
+    {:ok, lead_statuses}
+  end
+
+  @doc """
+  Gets the potential statuses of an opportunity from CloseIO.
+
+  Returns `{:ok, opportunity_statuses}`.
+
+  ## Examples
+
+    > Closex.MockClient.get_opportunity_statuses()
+    ...contents of test/fixtures/opportunity_statuses.json...
+  """
+  def get_opportunity_statuses(opts \\ []) do
+    opportunity_statuses = load("opportunity_statuses.json")
+    send self(), {:closex_mock_client, :get_opportunity_statuses, [opts]}
+    {:ok, opportunity_statuses}
+  end
+
+  @doc """
+  Gets the users available in CloseIO.
+
+  Returns `{:ok, users}`.
+
+  ## Examples
+
+    > Closex.MockClient.get_users()
+    ...contents of test/fixtures/users.json...
+  """
+  def get_users(opts \\ []) do
+    users = load("users.json")
+    send self(), {:closex_mock_client, :get_users, [opts]}
+    {:ok, users}
+  end
+
+  # TODO: implement these mocks
```

Comment:
This TODO is a bit outdated, can you apply it only to the `:noop` functions?

Reply:
As part of the swapout of the testing clients in our umbrella I've had to replace the `:noop` functions by delegating to the HTTPClient. I've updated the TODO to reflect this.